packages/stylist-macros/src/output/context_recorder.rs

```diff
@@ -41,12 +65,24 @@ impl ContextRecorder {
         Self::default()
     }
 
+    pub fn uses_nested(&mut self, other: &Self) {
+        self.usage = ExpressionKind::joint_kind(self.usage, other.usage);
+    }
+
     // Record the usage of a dynamic expression
     pub fn uses_dynamic_argument(&mut self) {
-        self.usage = AllowedUsage::Dynamic;
+        self.usage = ExpressionKind::joint_kind(self.usage, ExpressionKind::Dynamic);
+    }
+    // Record the usage of an expression that is not allowed in const context
+    pub fn uses_static(&mut self) {
+        self.usage = ExpressionKind::joint_kind(self.usage, ExpressionKind::Static);
     }
 
     pub fn is_static(&self) -> bool {
-        self.usage == AllowedUsage::Static
+        matches!(self.usage, ExpressionKind::Static | ExpressionKind::Const)
+    }
+
+    pub fn is_const(&self) -> bool {
+        matches!(self.usage, ExpressionKind::Const)
```

Comment:
`self.usage == ExpressionKind::Const`?

Reply:
Mostly so it matches the style used in `is_static`, I think both are mentally the same overhead, close to none.